src/app/services/app-auth.service.ts

```diff
@@ -0,0 +1,158 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { OAuthErrorEvent, OAuthService } from 'angular-oauth2-oidc';
+import { BehaviorSubject } from 'rxjs';
+
+/**
+ *
+ * This service is build with external OIDC compliant OAuthService
+ * for demonstration purposes, be careful and read all of the comments
+ *
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class AppAuthService {
+  private isAuthenticated$$ = new BehaviorSubject<boolean>(false);
+  public isAuthenticated$ = this.isAuthenticated$$.asObservable();
```

Comment:
```suggestion
  public isAuthenticated$ = this.oauthService.events.pipe(map(() => this.oauthService.hasValidAccessToken()));
```

Reply:
As for me, it looks like an improvement and can be applied.
Please, also move the comment from the constructor, this one:
`After successful auth and the `token_received` event it will update isAuthenticated$ to activate the routes`

Reply:
It does not let me apply it from here but I will add the change manually

Reply:
fixed